src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -175,6 +176,44 @@ private void AddExceptions(ISymbol symbol)
                 }
             }
 
+            protected abstract void AddCaptures(ISymbol symbol);
+
+            protected void AddCaptures(SyntaxNode syntax)
+            {
+                var semanticModel = GetSemanticModel(syntax.SyntaxTree);
+                if (semanticModel.IsSpeculativeSemanticModel)
+                {
+                    // Speculative semantic models cannot do region analysis
+                    return;
+                }
+
+                var analysis = semanticModel.AnalyzeDataFlow(syntax);
+                var captures = analysis.CapturedInside;
+                if (!captures.IsEmpty)
+                {
+                    var parts = new List<SymbolDisplayPart>();
+                    parts.Add(new SymbolDisplayPart(kind: SymbolDisplayPartKind.Text, symbol: null, text: $"\r\n{WorkspacesResources.Variables_captured_colon}"));
+                    bool first = true;
+                    foreach (var captured in captures)
+                    {
+                        if (!first)
+                        {
+                            parts.AddRange(Punctuation(","));
+                        }
+
+                        parts.AddRange(Space(count: 1));
+                        parts.AddRange(ToMinimalDisplayParts(captured, s_formatForCaptures));
```

Comment:
What kind is being used here for 'this'/'Me'? Keyword (so we get colorization) or no?

Reply:
The parts seem to have the right kinds (shown below). But it looks like parameters and locals are all rendered with the same color anyways (white in my theme).

![image](https://user-images.githubusercontent.com/12466233/35176033-9cdc6a8c-fd2b-11e7-9e47-bbd0c32722e8.png)
![image](https://user-images.githubusercontent.com/12466233/35176244-b677cf3a-fd2c-11e7-9249-be70027338eb.png)

Reply:
I guess I'd expect the magic this parameter to be called a keyword, not a parameter. Our symbol model indeed calls it parameter; users don't.